File: pkg/codegen/testing/utils/testdata/components-pp/dotnet/SimpleComponent.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Random = Pulumi.Random;

namespace Components
{
    public class SimpleComponent : global::Pulumi.ComponentResource
    {
        public SimpleComponent(string name, ComponentResourceOptions? opts = null)
            : base("components:index:SimpleComponent", name, ResourceArgs.Empty, opts)
        {
            var firstPassword = new Random.RandomPassword($"{name}-firstPassword", new()
            {
                Length = 16,
                Special = true,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            var secondPassword = new Random.RandomPassword($"{name}-secondPassword", new()
            {
                Length = 16,
                Special = true,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            this.RegisterOutputs();
        }
    }
}

File: pkg/codegen/testing/utils/testdata/components-pp/python/simpleComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_random as random

class SimpleComponent(pulumi.ComponentResource):
    def __init__(self, name: str, opts: Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:SimpleComponent", name, {}, opts)

        first_password = random.RandomPassword(f"{name}-firstPassword",
            length=16,
            special=True,
            opts = pulumi.ResourceOptions(parent=self))

        second_password = random.RandomPassword(f"{name}-secondPassword",
            length=16,
            special=True,
            opts = pulumi.ResourceOptions(parent=self))

        self.register_outputs()
